plugins/module_utils/foreman_helper.py

```diff
@@ -125,17 +125,29 @@ def find_resource(self, resource, search, params=None, failsafe=False, thin=Fals
             params = {}
         params['thin'] = thin
         results = self.list_resource(resource, search, params)
+        if resource == 'snapshots':
+            # Snapshots API does not do search
+            snapshot = []
+            data = re.search(r"name=\"(.*)\"", search)
```

Comment:
I very much dislike to use `re` to parse something we generated ourself in `find_resource_by_name`.

At this point I wonder if we should extend the `find_resource` API to take both, `name` and `search` (both defaulting to `None`) and move the `search` generation into it. Basically:

```python
def find_resource(self, resource, search='', name=None, params=None, failsafe=False, thin=False):
    if name is not None:
        search += ' name="{}"'.format(name)
    …
```

Reply:
(and then probably `search.strip()` to get rid of spaces)